src/COMPONENTS/WELCOMEPAGE/WelcomePage.tsx

```diff
@@ -0,0 +1,78 @@
+import { useEffect } from "react";
+import { Link, useNavigate } from "react-router-dom";
+import { useSelector, useDispatch } from "react-redux";
+// @ts-ignore
+import { setProfileDetails } from "../../REDUX/profileDetailSlice.js";
+
+import "./welcomepage.css";
+export const WelcomePage = () => {
+  const navigate = useNavigate();
+  const dispatch = useDispatch();
+  const { name, id, adminRights } = useSelector(
+    (state: any) => state.profileDetailSlice
+  );
+
+  const displayName = name || "Page";
+
+  useEffect(() => {
+    // @ts-ignore
+    const isLoggedIn = JSON.parse(localStorage.getItem("isLoggedIn"));
+
+    if (isLoggedIn) {
+      // @ts-ignore
+      const userData = JSON.parse(localStorage.getItem("loggedInUser"));
+      dispatch(setProfileDetails(userData));
+      navigate("/");
+    } else {
+      dispatch(setProfileDetails({}));
+    }
+  }, []);
+  useEffect(() => {
+    //initialize given data into local storage
```

Comment:
In my opinion we should store this default data in a `dataProvider` object, that gets referred to from `App.tsx`.  IMHO - it should not be declared in the welcome page.